qiskit/utils/deprecation.py

```diff
@@ -77,3 +92,73 @@ def _rename_kwargs(func_name, kwargs, kwarg_map, category: Type[Warning] = Depre
                 )
 
                 kwargs[new_arg] = kwargs.pop(old_arg)
+
+
+def _extend_docstring(func, version, kwarg_map):
+    """kwarg_map[None] means message in no kwarg, and it will be
+    append to the long desscription"""
+    docstr = func.__doc__
+    if docstr:
+        docstr_lines = docstr.expandtabs().splitlines()
+    else:
+        docstr_lines = ["DEPRECATED"]
+
+    indent = 1000
+    first_empty_line = None
+    for line_no, line in enumerate(docstr_lines[1:], start=1):
+        stripped = line.lstrip()
+        if stripped:
+            indent = min(indent, len(line) - len(stripped))
+        else:
+            if first_empty_line is None:
+                first_empty_line = line_no
+    if first_empty_line is None:
+        first_empty_line = len(docstr_lines)
+    spaces = ""
+    if indent != 1000:
+        spaces = " " * indent
+
+    new_doc_str_lines = docstr_lines[:first_empty_line]
+    if None in kwarg_map:
+        new_doc_str_lines += ["", spaces + f".. deprecated:: {version}"]
+        for msg_line in kwarg_map[None]:
+            new_doc_str_lines.append(spaces + "  " + msg_line)
+
+    arg_indent = indent
+    args_section = False
+    deprecated_arg = False
+    for docstr_line in docstr_lines[first_empty_line:]:
+        stripped = docstr_line.lstrip()
+        current_indent = len(docstr_line) - len(stripped)
```

Comment:
The exact value of the indentation of one line isn't enough to tell what block you're in; blank lines will always have an indentation of 0 but don't necessarily end a block, and continuation paragraph lines are generally more indented than the first one, but should still be seen as in the "Args" block.